model/instrument.py

```diff
@@ -0,0 +1,193 @@
+            if self.stream_status.allow_response_cycle == 99999999999:
+                self.stream_status.allow_response_cycle = self.stream_status.cycle \
+                    + context.agent_op.operator_response_delay(context)
+        elif random.random() < self.stream_status.p_stream_shift:
+            self.stream_status.stream_pos = round(self.stream_status.stream_pos + (
+                self.stream_status.stream_shift_amount * random.choice(
+                    [i for i in range(-1, 2) if i not in [0]])), 4)
+            if self.stream_status.allow_response_cycle == 99999999999:
+                self.stream_status.allow_response_cycle = self.stream_status.cycle \
+                    + context.agent_op.operator_response_delay(context)
+        context.messages.concatbegining(f"{self.show_pos()}\n{get_current_datapoints(context)}\n")
+        if self.stream_status.cycle >= self.stream_status.allow_response_cycle:
+            self.instrument_status.msg = self.instrument_status.msg + "<?>"
+            # Warning! WWW This used to truncate, but that interacts badly with computer math
+            # bcs occassionally you'll end up with 0.6999999 which truncation makes 0.6 instead
+            # of 0.7, and it loops out.
+        self.stream_status.cycle = self.stream_status.cycle + 1
+
+    def update(self, context:Context):
+        "update vars in relation to time"
+        if self.collecting_data:
+            self.stream_frame_update(context)
+            if self.run_start_frame:
+                self.run_start_frame = False
+            else:
+                context.messages.concat(f"Run {self.run_number} "+
+                    f"{colored('Collecting Data', 'green')}\n")
+            
+            current_sample = context.ami.samples[self.current_sample]
+            if len(current_sample.data) > self.time_out_value:
+                context.printer(f"{colored('Warning', 'yellow')}: To many data points timeout", "Warning: To many data points timeout")
+                context.agenda.add_event(context.instrument.run_number, context.instrument.run_start_time, context.current_time, True)
+                current_sample.timeout = True
+                self.collecting_data = False
+            else:
+                # Normal Distribution of quality based on how far away beam pos(user controller)
+                # is away from the stream pos (instrument beam)
+                # Farthest away is 2 beam on one side and stream on other
+                # Closest is 0 right on top of each other
+                # FFF: allow this to work with less than 1 second time steps
+                delta:timedelta = context.current_time - self.last_data_update
+                distance = abs(self.stream_status.stream_pos - self.beam_status.beam_pos)
+                for _ in range(int(delta.total_seconds()) * self.data_per_second):
+                    datapoint:DataPoint = DataPoint(clamp(aquire_data(distance,
```

Comment:
## Important!
This is where the calculations are for determining the value of the data point I don't think this is where the problem is but it might be a problem here too